src/transformers/models/mplug_owl/modeling_mplug_owl.py

```diff
@@ -0,0 +1,1574 @@
+# coding=utf-8
+# Copyright 2023 x-plug The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch MplugOwl model."""
+
+import math
+from typing import Any, Optional, Tuple, Union
+
+import torch
+
+from transformers.utils.import_utils import is_flashattention_available
+
+
+if is_flashattention_available():
+    from flash_attn.flash_attn_interface import flash_attn_unpadded_func as flash_attn_func
+
+
+from dataclasses import dataclass
+
+import torch.utils.checkpoint
+from torch import nn
+
+from transformers.activations import QuickGELUActivation
```

Comment:
Here you should probably import `ACT2FN` and use `config.hidden_act` to retrieve the correct activation function - similarly as: https://github.com/huggingface/transformers/blob/3658488ff77ff8d45101293e749263acf437f4d5/src/transformers/models/gpt2/modeling_gpt2.py#L351